### PR TITLE
Fix for upstream log4cxx

### PR DESCRIPTION
**Pull request by flixr**
Fixes empty log lines on the console with upstream version of log4cxx.

In upstream log4cxx the calls `log4cxx::Level::getDebug()`, etc. now return new pointers on every call.
See https://issues.apache.org/jira/browse/LOGCXX-394

This resulted in rosconsole `Formatter::print` being called with an invalid level (`levels::Count`)
and hence `color` was NULL and nothing printed (except the newline).

So instead of calling getDebug(), directly compare the integer level which works in the released version and upstream.
It should also be more efficient to directly compare integer level when the upstream version of log4cxx is used.

~~While here also change/fix console printer to print unknown levels.~~ see #34

### CODE REVIEW

File: src/rosconsole/impl/rosconsole_log4cxx.cpp [hunk 1]
```diff
@@ -84,27 +84,27 @@ class ROSConsoleStdioAppender : public log4cxx::AppenderSkeleton
   }
 
 protected:
-  virtual void append(const log4cxx::spi::LoggingEventPtr& event, 
+  virtual void append(const log4cxx::spi::LoggingEventPtr& event,
```
Comment (dirk-thomas) [inline src/rosconsole/impl/rosconsole_log4cxx.cpp]: Please revert unrelated whitespace changes like this.

File: src/rosconsole/impl/rosconsole_log4cxx.cpp [hunk 2]
```diff
@@ -379,7 +379,7 @@ void shutdown()
     g_log4cxx_appender = 0;
   }
   // reset this so that the logger doesn't get crashily destroyed
-  // again during global destruction.  
+  // again during global destruction.
```
Comment (dirk-thomas) [inline src/rosconsole/impl/rosconsole_log4cxx.cpp]: Please revert unrelated whitespace changes like this.

File: src/rosconsole/rosconsole.cpp [hunk 3]
```diff
@@ -362,8 +362,6 @@ void Formatter::print(void* logger_handle, ::ros::console::Level level, const ch
     color = COLOR_GREEN;
   }
 
-  ROS_ASSERT(color != NULL);
-
```
Comment (dirk-thomas) [inline src/rosconsole/rosconsole.cpp]: The changes in this file seems to be unrelated too. Please revert and consider to create a separate PR for it.

**Review by dirk-thomas — COMMENTED** [review]
With what versions of `log4cxx` is this change backwards compatible?

File: src/rosconsole/impl/rosconsole_log4cxx.cpp (hunk 1 above)
Comment (flixr) [reply src/rosconsole/impl/rosconsole_log4cxx.cpp]: reverted.

File: src/rosconsole/impl/rosconsole_log4cxx.cpp (hunk 2 above)
Comment (flixr) [reply src/rosconsole/impl/rosconsole_log4cxx.cpp]: reverted

File: src/rosconsole/rosconsole.cpp (hunk 3 above)
Comment (flixr) [reply src/rosconsole/rosconsole.cpp]: I came across this while fixing it for the upstream version to still print something.
But it is now in a separate PR: #34

**Comment by flixr** [comment]
This is tested and compatible with the released version 0.10.0 (from 2010) that is also in Debian/Ubuntu as well as the latest master from upstream.

**Comment by dirk-thomas** [comment]
Thank you for the contribution.
